fuzzers/engine.ts

```diff
@@ -0,0 +1,156 @@
+import { IFuzzer } from 'fuzzer';
+import os from 'os';
+import readline from 'readline';
+import { isMainThread, parentPort, Worker as ThreadWorker, workerData } from 'worker_threads';
+
+enum WorkerMessageTypes {
+	Online = 'online',
+	Progress = 'progress',
+	Crash = 'crash',
+}
+
+/**
+ * Multi-threaded fuzzing engine.
+ */
+export default class Engine<TFuzzer extends IFuzzer> {
+	run(fuzzer: TFuzzer, filename: string): void {
+		if (isMainThread) {
+			this.run_main(filename);
+		} else {
+			this.run_worker(fuzzer);
+		}
+	}
+
+	private run_main(filename: string): void {
```

Comment:
Let's keep to the codestyle and call this runMain. We do not really use underscores in this repo

Reply:
You're right, sloppy, too much Rust lately ;)